test/built-ins/Set/prototype/intersection/set-like-class-order.js

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2023 Anthony Frehner and Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set.prototype.intersection
+description: Set.prototype.intersection calls a Set-like class's methods in order
+features: [set-methods]
+includes: [compareArray.js]
+---*/
+
+let observedOrder = [];
+
+function observableIterator() {
```

Comment:
Here's where I wish we had already expanded the Observable utilities in `harness/temporalHelpers.js` into their own, more general, helper file ... oh well, next time :smile: